### PR TITLE
Clean up GS.h includes and beef up the PCH

### PR DESCRIPTION
**Pull request by Ziemas**
Hopefully cleaning up GS.h help build times a bit.

Adding stuff to the PCH will make the No-PCH builds even worse though. Let's consider if it's worth it.

### CODE REVIEW

**Comment by lightningterror** [comment]
Would be nice to properly name the first commit.

**Comment by Ziemas** [comment]
Done

File: pcsx2/GS/Renderers/Common/GSDevice.h [hunk 1]
```diff
@@ -21,6 +21,11 @@
 #include "GSVertex.h"
 #include "GS/GSAlignedClass.h"
 #include "GSOsdManager.h"
+#include <array>
+#ifdef _WIN32
+#include <dxgi1_3.h>
+#endif
```
Comment (tadanokojin) [inline pcsx2/GS/Renderers/Common/GSDevice.h:27]: there some reason you stuck this here instead of device11?

Comment (Ziemas) [reply pcsx2/GS/Renderers/Common/GSDevice.h:27]: No probably not, I think I just slapped includes around until it built.

Can move it.

**Comment by Ziemas** [comment]
Obsoleted by #4709

**Pull request closed without merging**